tests/unit/aspect.ts

```diff
@@ -171,6 +171,41 @@ registerSuite({
 			}
 		},
 
+		'.on()': {
+			'advising function returns undefined, returns original result'() {
+				const aspectStub1 = sinon.stub();
+
+				aspect.on(obj, 'method', aspectStub1);
+
+				assert.strictEqual(obj.method(0), 1);
+				assert.isTrue(aspectStub1.calledAfter(methodSpy));
+			},
+
+			'advising function returns defined values, returns advising function result'() {
+				const aspectStub1 = sinon.stub().returns(2);
+
+				aspect.on(obj, 'method', aspectStub1);
+
+				assert.strictEqual(obj.method(0), 2);
```

Comment:
Ok, I have clearly missed something... there is no `aspect#on` in Dojo 1.  Why do we need `core/aspect#on` in Dojo 2 when we also have `core/on`?  Won't that lead to a significant amount of confusion (like it just did for me?)  Which do I use, when do I use it, why doesn't it work?  It will be `dojo.byId` and `dijit.byId` all over again!

I took a look at other AOP implementations and I don't see any `on` advice.  I have seen only `before`, `after`, and `around` with some variations like "after throwing" and "after finally".  `aspect#on` might imply (thought it doesn't technically do this at the moment) "after finally" so wouldn't it be better to call it `afterFinally`?